### ISSUE TITLE
Admin pagination issues with named servers

### ISSUE THREAD

**Issue by rcthomas** [jupyterhub/jupyterhub#3334]
### Bug description

Pagination's great, but I think it gets confused by named servers.  After named servers are registered they count as "users" (entries?) when being rendered, but the `pagination.total` is only counting users. 

#### Expected behaviour

If you have named servers going on, maybe the admin page should set the total to be from the total number of `Spawner` entries or take that into account.  The behavior for no named servers could be just left alone.  The admin page could say it was listing spawners in the bottom too.

#### Actual behaviour

Assuming default settings, you might have 789 users, but if they have some named servers you won't be able to see them all listed unless you override the `page` argument. If each one has a named server, you'll only see like half. Interestingly though you'll see at least the number of default entries per page (like if a user has 105 named servers you'll see their user row and 105 named server rows, but no other users).

### How to reproduce

1. Set up a hub with an admin and test users, say 12 of them
2. For `jupyterhub_config.py`:

```
c.JupyterHub.admin_users = set(["admin"])
c.JupyterHub.allow_named_servers = True
c.JupyterHub.default_url = "/hub/home"
c.Authenticator.allowed_users = set([
    "user01", "user02", "user03",
    "user04", "user05", "user06",
    "user07", "user08", "user09",
    "user10", "user11", "user12",
    "admin"
])
c.Pagination.default_per_page = 3
```

3. Login as the `admin` user and go to the admin page. You should see 3 users listed and pages for 5 pages worth of users which seems right.
4. Go to the home page and add say 4 servers for yourself.
5. When you're done, go back to the admin page.  What you see is the admin user, and then the 4 servers you created.  The pagination info still says it's showing a subset of users for 13 total users and there are 5 pages worth
6. Move to the last page; some of the higher-numbered users will be missing.
7. They are still there but you have to set `?page=` off the end of the list of expected pages.

### Work

I'm going to see about whether the proposed solution works and submit a PR but I'm a little confused how the template knows to stop showing users past the per_page setting.  If anyone has a hint let me know but I'll probably eventually track it down.

**Comment by rcthomas** [comment]
Looks like you just need to make the user query return only distinct users then apply the limit and offset.  PR shortly.

**Comment by rcthomas** [comment]
Closed by #3335
